aclk/aclk_query.c

```diff
@@ -537,16 +537,30 @@ void *aclk_query_main_thread(void *ptr)
     }
 
     while (!netdata_exit) {
+        if(aclk_disable_runtime) {
+            sleep(1);
+            continue;
+        }
         ACLK_SHARED_STATE_LOCK;
-        if (unlikely(!aclk_shared_state.metadata_submitted)) {
-            ACLK_SHARED_STATE_UNLOCK;
+        if (unlikely(!aclk_shared_state.version_neg)) {
+            if (!aclk_shared_state.version_neg_wait_till || aclk_shared_state.version_neg_wait_till > now_monotonic_usec()) {
+                ACLK_SHARED_STATE_UNLOCK;
+                info("Waiting for ACLK Version Negotiation message from Cloud");
+                sleep(1);
+                continue;
+            }
+            error("ACLK version negotiation failed. No reply to \"hello\" with \"version\" from cloud in time of %ds. "
```

Comment:
You can set  `errno = 0`  o avoid misleading info e.g.

```
ACLK version negotiation failed. No reply to "hello" with "version" from cloud in time of 3s.  Reverting to default ACLK version of 1. (errno 22, Invalid argument)
```

Reply:
thanks, will change.
I am thinking though since this is repeating a lot -> should we have some `error` equivalet that ignores `errno` to avoid having to set it to 0 in bazzilion places in the code...

Reply:
`3s.  Reverting` removed the double space as well :D